pkg/monitoring/metrics/virt-handler/domainstats/guest_agent_info_scraper_test.go

```diff
@@ -0,0 +1,221 @@
+			scraper.mutex.RLock()
+			cached, exists := scraper.cache[socketFile]
+			withinTimeout := exists && time.Since(cached.timestamp) < cacheTimeout
+			scraper.mutex.RUnlock()
+
+			Expect(withinTimeout).To(BeTrue())
+		})
+
+		It("should identify cache entries outside timeout", func() {
+			By("creating an expired cache entry")
+			scraper.mutex.Lock()
+			scraper.cache[socketFile] = &guestAgentInfoCache{
+				timestamp: time.Now().Add(-2 * cacheTimeout),
+				info:      guestInfo.DeepCopy(),
+			}
+			scraper.mutex.Unlock()
+
+			By("checking if entry is outside timeout")
+			scraper.mutex.RLock()
+			cached, exists := scraper.cache[socketFile]
+			withinTimeout := exists && time.Since(cached.timestamp) < cacheTimeout
+			scraper.mutex.RUnlock()
+
+			Expect(withinTimeout).To(BeFalse())
+		})
+	})
+
+	Context("Cache cleanup", func() {
+		It("should clean up expired cache entries", func() {
+			By("adding both fresh and expired entries to cache")
+			scraper.mutex.Lock()
+			scraper.cache[socketFile] = &guestAgentInfoCache{
+				timestamp: time.Now().Add(-2 * cacheTimeout), // Expired
+				info:      guestInfo.DeepCopy(),
+			}
+			freshSocketFile := "/tmp/fresh-socket"
+			scraper.cache[freshSocketFile] = &guestAgentInfoCache{
+				timestamp: time.Now(), // Fresh
+				info:      guestInfo.DeepCopy(),
+			}
+			scraper.mutex.Unlock()
+
+			By("verifying both entries exist before cleanup")
+			scraper.mutex.RLock()
+			_, expiredExists := scraper.cache[socketFile]
+			_, freshExists := scraper.cache[freshSocketFile]
+			scraper.mutex.RUnlock()
+			Expect(expiredExists).To(BeTrue())
+			Expect(freshExists).To(BeTrue())
+
+			By("calling cleanup")
+			scraper.cleanupExpiredCache()
+
+			By("verifying only expired entry is removed")
+			scraper.mutex.RLock()
+			_, expiredExists = scraper.cache[socketFile]
+			_, freshExists = scraper.cache[freshSocketFile]
+			scraper.mutex.RUnlock()
+
+			Expect(expiredExists).To(BeFalse())
+			Expect(freshExists).To(BeTrue())
+		})
+
+		It("should not clean up non-expired cache entries", func() {
+			By("adding a fresh cache entry")
+			scraper.mutex.Lock()
+			scraper.cache[socketFile] = &guestAgentInfoCache{
+				timestamp: time.Now(),
+				info:      guestInfo.DeepCopy(),
+			}
+			scraper.mutex.Unlock()
+
+			By("calling cleanup immediately")
+			scraper.cleanupExpiredCache()
+
+			By("verifying cache entry still exists")
+			scraper.mutex.RLock()
+			_, exists := scraper.cache[socketFile]
+			scraper.mutex.RUnlock()
+			Expect(exists).To(BeTrue())
+		})
+
+		It("should handle empty cache during cleanup", func() {
+			By("ensuring cache is empty")
+			Expect(scraper.cache).To(BeEmpty())
+
+			By("calling cleanup on empty cache")
+			Expect(func() {
+				scraper.cleanupExpiredCache()
+			}).ToNot(Panic())
+
+			By("verifying cache is still empty")
+			Expect(scraper.cache).To(BeEmpty())
+		})
+	})
+
+	Context("Deep copy behavior", func() {
+		It("should store a deep copy of guest info in cache", func() {
```

Comment:
Not sure about the value of this test. `DeepCopy` is generated could be tested without going through the cache but I don't think we need to do that.